sdk/tables/Azure.Data.Tables/src/TableTransactionalBatch.cs

```diff
@@ -300,7 +300,7 @@ private bool AddMessage(ITableEntity entity, HttpMessage message, RequestType re
         /// Builds an ordered list of <see cref="HttpMessage"/>s containing the batch sub-requests.
         /// </summary>
         /// <returns></returns>
-        private List<(ITableEntity entity, HttpMessage HttpMessage)> BuildOrderedBatchRequests()
+        internal virtual List<(ITableEntity entity, HttpMessage HttpMessage)> BuildOrderedBatchRequests()
```

Comment:
Are you using this method in test? Why does it need to be proxied?

Reply:
the result of the method is used [here](https://github.com/Azure/azure-sdk-for-net/blob/01793bf2b016e410bda508034128e7172cdc07a8/sdk/tables/Azure.Data.Tables/src/TableTransactionalBatch.cs#L222). Before this was proxied `_submittedMessageList` would be null.

Reply:
And it is referenced by the test [here](https://github.com/Azure/azure-sdk-for-net/blob/01793bf2b016e410bda508034128e7172cdc07a8/sdk/tables/Azure.Data.Tables/tests/TableClientLiveTests.cs#L1100)

Reply:
The call at https://github.com/Azure/azure-sdk-for-net/blob/01793bf2b016e410bda508034128e7172cdc07a8/sdk/tables/Azure.Data.Tables/tests/TableClientLiveTests.cs#L1100 should be itself proxied to the base

Reply:
You are correct, I only needed `TryGetFailedEntityFromException to be virtual.